### PR TITLE
ISPN-11275 expose local site name

### PR DESCRIPTION
**Pull request by karesti**
https://issues.redhat.com/browse/ISPN-11275

backport #7864

### CODE REVIEW

**Comment by karesti** [comment]
not sure at all about this

File: core/src/main/java/org/infinispan/manager/CacheManagerInfo.java [hunk 1]
```diff
@@ -107,6 +108,11 @@ public String getClusterName() {
       return configurationManager.getGlobalConfiguration().transport().clusterName();
    }
 
+   public String getLocalSite() {
+      SiteConfiguration sites = configurationManager.getGlobalConfiguration().sites();
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/manager/CacheManagerInfo.java]: IMO, you should get the site name from RELAY2. the site-name in global configuration isn't mandatory to set and it may be diffeent.

Comment (gustavonalle) [reply core/src/main/java/org/infinispan/manager/CacheManagerInfo.java]: +1, this seems a random arbitrary property someone left behind :)
Also the infinispan-xsite.xml sample doc does not use it either

Comment (pruivo) [reply core/src/main/java/org/infinispan/manager/CacheManagerInfo.java]: created:https://issues.redhat.com/browse/ISPN-11278

File: server/rest/src/test/java/org/infinispan/rest/resources/CacheManagerResourceTest.java [hunk 2]
```diff
@@ -213,6 +212,7 @@ public void testInfo() throws Exception {
       assertFalse(cmInfo.get("version").asText().isEmpty());
       assertEquals(2, cmInfo.get("cluster_members").size());
       assertEquals(2, cmInfo.get("cluster_members_physical_addresses").size());
+      assertTrue(cmInfo.get("local_site").isNull());
```
Comment (pruivo) [inline server/rest/src/test/java/org/infinispan/rest/resources/CacheManagerResourceTest.java]: I'm not sure how useful this test is :)

Comment (karesti) [reply server/rest/src/test/java/org/infinispan/rest/resources/CacheManagerResourceTest.java]: Test that the property is there with than name. Rest api contract

File: server/rest/src/test/java/org/infinispan/rest/resources/CacheManagerResourceTest.java [hunk 3]
```diff
@@ -1,20 +1,8 @@
...
-import static org.testng.AssertJUnit.assertEquals;
-import static org.testng.AssertJUnit.assertFalse;
-
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.stream.Collectors;
-
+import com.fasterxml.jackson.databind.JsonNode;
```
Comment (pruivo) [inline server/rest/src/test/java/org/infinispan/rest/resources/CacheManagerResourceTest.java]: revert the changes in the imports

File: server/rest/src/test/java/org/infinispan/rest/resources/CacheManagerResourceTest.java (hunk 2 above)
Comment (pruivo) [reply server/rest/src/test/java/org/infinispan/rest/resources/CacheManagerResourceTest.java]: well, it doesn't test if it returns the correct name... I'll close my eyes :)

File: server/rest/src/test/java/org/infinispan/rest/resources/CacheManagerResourceTest.java (hunk 3 above)
Comment (karesti) [reply server/rest/src/test/java/org/infinispan/rest/resources/CacheManagerResourceTest.java]: reverted

File: server/rest/src/test/java/org/infinispan/rest/resources/CacheManagerResourceTest.java (hunk 2 above)
Comment (karesti) [reply server/rest/src/test/java/org/infinispan/rest/resources/CacheManagerResourceTest.java]: is there an easy way to configure the local site name for the test?

Comment (pruivo) [reply server/rest/src/test/java/org/infinispan/rest/resources/CacheManagerResourceTest.java]: use the method that accepts the `TransportFlags` when instantiating the `CacheManager` and  use `TransportFlags.minimalXsiteFlags()`
The local site name will be `LON-1`

**Comment by pruivo** [comment]
integrated! thanks @karesti !
